### ISSUE TITLE
Add option to used cached firmware/skip online check

### ISSUE THREAD

**Issue by swests** [ruimarinho/shelly-updater#10]
Hi

My network is set like this:
VLAN 0 (LAN) - full access
VLAN 10 (IoT) - Internet of Things - devices can access internet and one specific host on LAN only
VLAN 20 (NoT) - Network of Things - devices can only access one host on LAN only.

1. If I run mota on my laptop or other server when on LAN, it can interrogate the shelly but the shelly can't call-back to mota to get the firmware (says success up update doesn't happen), because of my firewall rules.
2. If I switch my laptop to the NoT network, mota fails because it can't get to the internet to check for firmware.

If I run mota on my laptop on main LAN I see it downloads and caches the files, but as per 1. above, the update would fail.  If I could then (manually) switch network and re-run mote with a --use-cache or --no-download flag, I would be able to update.

I'd rather not open up my NoT network and lots of devices will then all call home.

**Comment by frankforpresident** [comment]
I have a similar setup and I have added a new firewall rule to allow communication between my laptop IP on LAN and NoT. When the upgrade is done I disable the rule.

fyi, when you are running on mac I would check this issue too. Might prevent some headaches
https://github.com/ruimarinho/mota/issues/8

**Comment by swests** [comment]
Yes, @frankforpresident  - I do this today. This and switching network are equally inconvenient, but the firewall-rule option has the potential to be left open, where as switching networks, I think, presents less of a security hole.

**Comment by frankforpresident** [comment]
Rather forgetting one IP address than opening up my network but I get what you mean 😄

**Comment by ruimarinho** [comment]
Still a valid feature request. v3.0.0 changes discovery from zeroconf to HTTP subnet scanning, but firmware files are still downloaded from Shelly's CDN.

One thing I'm not sure about is how the interface would work in practice — each device model has its own unique firmware URL on Shelly's CDN. Would you envision something like a `--firmware-dir` flag that points to a local directory of pre-downloaded `.zip` files? Or a `--offline` mode that only uses whatever is already in the cache (`~/.cache/com.github.ruimarinho.mota/`)? Open to suggestions on what would fit your workflow best.
